src/AutoRest.CSharp/Common/Output/PostProcessing/PostProcessor.cs

```diff
@@ -179,8 +179,16 @@ private async Task<Project> RemoveMethodsFromModelFactoryAsync(Project project,
             return project;
 
         var root = await modelFactoryGeneratedDocument.GetSyntaxRootAsync();
-        root = root?.RemoveNodes(nodesToRemove, SyntaxRemoveOptions.KeepNoTrivia);
-        modelFactoryGeneratedDocument = modelFactoryGeneratedDocument.WithSyntaxRoot(root!);
+        Debug.Assert(root is not null);
```

Comment:
Under what kind of circumstances would it reach this assert error?

Reply:
that depends on how Roslyn behaves, I am not 100 percent sure but when the code in `modelFactoryGeneratedDocument` cannot compile (maybe it needs to be non-compilable in a specific way), it will return   null here.
In most cases with compilable code, it will never be null.